pkg/aspect/accessLogsManager.go

```diff
@@ -31,153 +31,117 @@ type (
-		return nil
+	if cfg.Log == nil {
+		ce = ce.Appendf("Log", "an AccessLog entry must be provided")
+		// We can't do any more validation without a Log
+		return
 	}
-	tmplStr := cfg.CustomLogTemplate
-	if _, err := template.New("test").Parse(tmplStr); err != nil {
-		return ce.Appendf("CustomLogTemplate", "could not parse template string: %v", err)
+	if cfg.Log.LogFormat == aconfig.ACCESS_LOG_FORMAT_UNSPECIFIED {
+		ce = ce.Appendf("Log.LogFormat", "a log format must be provided")
 	}
-	return nil
+
+	// TODO: validate custom templates when users can provide us with descriptors
+	return
 }
 
 func (e *accessLogsWrapper) Close() error {
 	return e.aspect.Close()
 }
 
 func (e *accessLogsWrapper) Execute(attrs attribute.Bag, mapper expr.Evaluator, ma APIMethodArgs) Output {
-	// TODO: would be nice if we could use a mutable.Bag here and could pass it around
-	// labels holds the generated attributes from mapper
-	labels := make(map[string]interface{})
-	for attr, e := range e.inputs {
-		if val, err := mapper.Eval(e, attrs); err == nil {
-			labels[attr] = val
-		}
-		// TODO: throw error on failed mapping?
-	}
-
-	// TODO: better way to ensure timestamp is available if not supplied
-	// in Report() requests.
-	if _, found := labels["timestamp"]; !found {
-		labels["timestamp"] = time.Now()
-	}
-
-	entry := adapter.LogEntry{
-		LogName: e.logName,
-		Labels:  make(map[string]interface{}),
-	}
-
-	for _, a := range e.attrNames {
-		if val, ok := labels[a]; ok {
-			entry.Labels[a] = val
-			continue
-		}
-		if val, found := attribute.Value(attrs, a); found {
-			entry.Labels[a] = val
-		}
-
-		// TODO: do we want to error for attributes that cannot
-		// be found?
+	labels, err := evalAll(e.labels, attrs, mapper)
+	if err != nil {
+		return Output{Status: status.WithError(fmt.Errorf("failed to eval labels for log %s with err: %s", e.name, err))}
 	}
 
-	if len(entry.Labels) == 0 {
-		// don't write empty access logs
-		return Output{Status: status.OK}
+	templateVals, err := evalAll(e.templateExprs, attrs, mapper)
+	if err != nil {
+		return Output{Status: status.WithError(fmt.Errorf("failed to eval template expressions for log %s with err: %s", e.name, err))}
 	}
 
 	buf := pool.GetBuffer()
-	if err := e.template.Execute(buf, entry.Labels); err != nil {
-		return Output{Status: status.WithError(err)}
+	if err := e.template.Execute(buf, templateVals); err != nil {
+		pool.PutBuffer(buf)
+		return Output{Status: status.WithError(fmt.Errorf("failed to execute payload template for log %s with err: %s", e.name, err))}
 	}
-	entry.TextPayload = buf.String()
+	payload := buf.String()
 	pool.PutBuffer(buf)
+
+	entry := adapter.LogEntry{
+		LogName:     e.name,
+		Labels:      labels,
+		TextPayload: payload,
+	}
 	if err := e.aspect.LogAccess([]adapter.LogEntry{entry}); err != nil {
-		return Output{Status: status.WithError(err)}
+		return Output{Status: status.WithError(fmt.Errorf("failed to log %s with err: %s", e.name, err))}
```

Comment:
How about "failed to report to %s with..."? I don't like the other verbs since we don't know if we're writing or appending or making a telephone call to Bob or something else entirely.